File: lib/src/models/call_status.dart
enum CallStatus {
  ringing,
  answering,
  answered,
  declined,
  cancelled,
  timeout,
  missed;

  static CallStatus fromName(String value) {
    return CallStatus.values.firstWhere(
      (status) => status.name == value,
      orElse: () => CallStatus.ringing,
    );
  }
}

File: android/src/main/kotlin/com/example/call_notification_sdk/model/CallStatus.kt
package com.example.call_notification_sdk.model

enum class CallStatus {
    ringing,
    answering,
    answered,
    declined,
    cancelled,
    timeout,
    missed;

    companion object {
        fun fromString(value: String?): CallStatus? {
            if (value.isNullOrEmpty()) return null
            return try {
                valueOf(value)
            } catch (_: IllegalArgumentException) {
                null
            }
        }
    }
}

